### ISSUE TITLE
[Bug] resolve returns None on second call with `maximum_ttl=0`

### ISSUE THREAD

**Issue by SeverinAlexB** [pubky/pkarr#105]
Calling resolve 2 times in a row with `maximum_ttl(0)` results in `None` consistently on the second call. See example.

```
    #[test]
    fn ttl_0_test() {
        let uri = "7fmjpcuuzf54hw18bsgi3zihzyh4awseeuq5tmojefaezjbd64cy";
        let pubkey: PublicKey = uri.try_into().unwrap();
        let client = PkarrClientBuilder::default().maximum_ttl(0).build().unwrap();

        // First Call
        let value = client.resolve(&pubkey);
        match value.unwrap() {
            Some(val) => println!("Found1!"),
            None => println!("None1"),
        }
        sleep(Duration::from_secs(1));


        // Second Call
        let value = client.resolve(&pubkey);
        match value.unwrap() {
            Some(val) => println!("Found2!"),
            None => println!("None2"),
        }
        
    }
```

Result is always the same:

```
Found1!
None2
```

pkarr v2.2.0

**Comment by Nuhvi** [comment]
Managed to replicate this in v2.2.0, but not in [v3](https://github.com/pubky/pkarr/tree/v3-snapshot) ... so not sure if I should spend the time fixing it in v2.2.0

**Comment by SeverinAlexB** [comment]
If it's fixed on v3 then all good.

**Comment by SeverinAlexB** [comment]
Ok, need to reopen this issue. pkdns doesn't work without it. As soon as the cached value is expired, pkarr always returns None. I need a fix for v2.2.0.
